### DISCUSSION TITLE
Figma design ideas for UX/UI

### DISCUSSION THREAD

**Discussion by BenGWeeks** [damus-io/damus#222] [Ideas]
Hi all,

It would be great to have some input into some UX/UI design ideas. I've been working with a great designer (Sasha) who amazingly put these Figma designs together in just a couple of days.

You can see the Figma designs here:

https://www.figma.com/proto/ORaT1T0Ywfbm0sIjwy5Rgq/Damus-UX?node-id=1%3A2&scaling=scale-down&page-id=0%3A1&starting-point-node-id=29%3A7210

Password: **keypair**
(If you log in, you can leave the most welcome comments)

Very much a work in progress, but I have submitted some pull requests that start implementing the more generic parts of this (namely #191, #196, #203 and #220). It is a good idea to have a consensus on some more "contentious" ideas within it before starting to submit pull requests on the rest.

Particularly interested in some of the ideas around tipping and incentivizing good actors in the system and also incentivizing the relays themselves. For example,

- How much Lightning integration should there be (such as tipping posts)?
- Above dependent, does the app need an integrated Lightning wallet for the best UX (right away or in the future)?
- How much do we want it to be like/dislike Twitter?

Cheers,

Ben

**Comment by olheadinternet** [comment]
i really like the invoices reflecting as paid, saves a trip out of the app.

__________________

when i see a number next to the speech bubble, my reflex was this would expand to comments... i do see this is what twitter does now, i am used to that being a ↪️ reply button, so my reflex should of been that this will open a reply dialog, but it wasn't, could just be me.

________________

on the topic of sats... maybe on expanding a note, it could be reflected as both the amount of sats, but also the number of individual contributions. jack doesn't need the sats, but couldn't i just give myself 20000 sats, at least if it was 20000/1 then it would be clear one person loves this, where as 100/90 would mean 90 people on average contributed 1.11 sats but 90 people care about this.

on the timeline:

🗨️2    🔁    🤙20    ⚡100

I think the shaka should stay, and likes still able to be given for free. on expand one would see 

🗨️2    🔁    🤙20    ⚡100/10

and those 10⚡ are apart of the 20🤙 and still are broadcasted as simple kind 7 reactions.

update: i've now come to understand that a simple kind 7 is just a +... maybe shaka could turn into ++ so all kind 7's appear like...

on the timeline:
🗨️2    🔁    🤙10    ⚡100
on expand:
🗨️2    🔁    ++30    ⚡100/10
which express the 10 shakas, 20 hearts from yosup.app and 10 lightning bolts that got sent as both sats and a kind 7 lightning bolt emoji. this way a client who only shows emoji with no payment option could still see 10 people lightning bolt emoji'd, communicating the amount of people who tipped.
_______________

Does the app need an integrated Lightning wallet for the best UX?

as someone who didn't have a ln wallet before joining nostr. From the nostr git, it first told me to go to anigma, the autogen wallet was a bit of a turn off. (what follows might not be the best practices or the right terms but pleb tried)

maybe damus could replace zeus, where you can add a... LNDhub? and new users can do this at their own pace once they've seen the fun and sats that can be had. fountain.fm exists on the app store, so apple seems to be ok with it being apart of an app. "Yet another wallet" was my feeling with fountain.fm. You will turn off users who come for censor resistant media by auto-generating a wallet. Those who have experience with ln will find the task of linking an existing wallet something fun to teach others. Make all the wallets better by making them want to be as interoperable and feature rich with damus and other clients. Some new people will only be comfortable with cash app, they probably already have an account, and hopefully they step up to the plate and provide the tools to "export?" their wallet into damus.
______________

i filed this enhancement request about post/post replies https://github.com/damus-io/damus/issues/139 it also mentions a proposal for separating chatter... chatter is notes that appear in channels, so if I wanted to be more conversational in a chat thread it wont make notes & replies less useful. If i wanted to discover people to follow, i might check who you reply to... if i wanted to discover channels to join, i would go see posts, replies, & chatter. maybe there could be a sub-list of channels one has joined and spoken in in this tab.


would look like...

Following | Follwers | Channels | Relays

then default the profiles to notes and replies
swipe left to reveal just notes
swipe right to reveal notes replies and chatter
maybe a brief transparent overlay that explains this, with a setting to turn off tutorial.

______________
just my 2 sats...
im gonna revisit the figma later and see if i have other ideas.

**Comment by n0rthranger** [comment]
Hi. I think maybe add right or left swipe to show a drop down menu where you can go to settings or edit profile.

**Reply by olheadinternet** [reply]
i agree, there needs to be a place to put extra stuff until it has a home in the ui
![SSCustomSideMenu-3376080397](https://user-images.githubusercontent.com/56426913/210388575-683f08ba-dd17-458c-b49b-3c0f2650c028.png)

**Comment by robagreda** [comment]
Hey, I got a design for Damus client as well, you can check it out here: [https://www.figma.com/file/G4n8Fh3CNgU3Znu2Uhdt1W/Damus-App?node-id=0%3A1&t=T4gg8YvO00gWhwWD-1](https://www.figma.com/file/G4n8Fh3CNgU3Znu2Uhdt1W/Damus-App?node-id=0%3A1&t=T4gg8YvO00gWhwWD-1). Would be great if we can collaborate since I've put a lot of effort on this and already looking nice. 🙏🏼 cheers

**Reply by robagreda** [reply]
here are some images for reference:

<img width="430" alt="Feed@2x" src="https://user-images.githubusercontent.com/3373850/210596495-0f9660cd-f902-41ab-9dbe-ae10ffbbfd36.png">
<img width="430" alt="DM@2x" src="https://user-images.githubusercontent.com/3373850/210596480-7455dfa0-e257-4d22-b3fe-52c85279a140.png">
<img width="430" alt="New Post@2x" src="https://user-images.githubusercontent.com/3373850/210596508-704bd601-be74-4c07-aca3-5fbca9e81553.png">
<img width="430" alt="Settings@2x" src="https://user-images.githubusercontent.com/3373850/210596513-03f187fd-7a17-4e5e-a025-34744d300750.png">

**Reply by olheadinternet** [reply]
I really like how you did the relay list with a bit more context to the red yellow green dots - was never sure if yellow was connecting or read only

**Reply by robagreda** [reply]
Nice feedback! I love it! I think there are a bunch of things within Damus that are a little bit technical and for a typical user are hard to understand sometimes, that's why UX is important. I am glad you liked that one. Let's see what @BenGWeeks and @jb55 say! I am excited about this project!

**Comment by olheadinternet** [comment]
Themes.

The following would be a bad idea to ship right away, but I thought I would throw it out there as if it’s a direction damus might take in the future, it’s probably easier to think about it sooner than later.

If this does get built in now, I don’t think it should ship until a year out from App store approval… it’s a good thing for damus that early screen shots of nostr all look the same If they come from damus.

With that out of the way, I’ve heard people suggest that themes (presumably a json file) is something they would like to have on a client. Think change the colors scheme of nip badges and hyperlinks, switch the order of menus around (flip dm and search, or put profile where dm is), maybe one could configure to switch the default kind 7 message sent. I would shy away from loading graphics in the background of the timeline or anything that would completely destroy the aesthetic. I don’t think these examples matter rather having a config.json  or theme.json file so that as features are designed that there is a place to put some config options if and when damus would want to open up theming it could do so easily.

**Comment by BenGWeeks** [comment]
Nice work. I personally prefer the "off black" look (dark grey) as you have it, but it's @jb55's baby so open to his feedback.

I agree with the left column for profile and signing out etc. Happy to implement.

DM's definitely need improving from current! :-)

**Reply by robagreda** [reply]
Your call man! I am willing to collaborate with you on this task!
